tests/arch/common/timing/testcase.yaml

```diff
@@ -0,0 +1,17 @@
+common:
+  filter: CONFIG_ARCH_HAS_TIMING_FUNCTIONS
+tests:
+  arch.common.timing:
+    filter: CONFIG_MP_NUM_CPUS == 1 and not CONFIG_QEMU_TARGET
+  arch.common.timing.smp:
+    filter: CONFIG_MP_NUM_CPUS > 1 and not CONFIG_QEMU_TARGET
+    extra_configs:
+      - CONFIG_SCHED_CPU_MASK=y
+  arch.common.timing.build_only:
+    filter: CONFIG_MP_NUM_CPUS == 1
+    build_only: true
```

Comment:
why build only?

Reply:
```As timing and emulation are a somewhat dangerous combination,
`testcase.yaml` will only build the tests if CONFIG_QEMU_TARGET is
configured and run them only if not.
```
This just adds un-necessary build variants of the same code. we need a better way to deal with this.

Reply:
The easy one here is to not run the build_only variant for non emulator targets (so there's no overlap of builds). Will update.

Reply:
After inspecting CI results, it seems that by having this `build_only`, just 5 emulator builds are gained, for architectures that are already built otherwise, so I'll update the PR to drop the `build_only` variant.